### PR TITLE
Initial appveyor configuration

### PR DESCRIPTION
**Pull request by Twon**
Appveyor build for Visual Studio 2019

### CODE REVIEW

File: linear_algebra/code/CMakeLists.txt [hunk 1]
```diff
@@ -48,30 +48,15 @@ include_directories(include test .)
...
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
-
-    set(CMAKE_C_FLAGS_RELEASE   "-O3 -march=westmere -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=westmere -std=c++17 -stdlib=libc++ -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
-
-elseif(CMAKE_CXX_COMPILER STREQUAL g++)
-
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra  -Wno-unused-function -Wno-unused-variable -W -Wno-unused-but-set-variable -fmax-errors=10")
```
Comment (Twon) [inline linear_algebra/code/CMakeLists.txt]: We setting C++ 17 compiler flags via the target properties which will do the right thing for every compiler.  No need to set the flags for compilers directly.

File: linear_algebra/code/CMakeLists.txt [hunk 2]
```diff
@@ -48,30 +48,15 @@ include_directories(include test .)
...
-    set(CMAKE_CXX_COMPILER clang++)
-else()
-    set(CMAKE_C_COMPILER   gcc)
-    set(CMAKE_CXX_COMPILER g++)
-endif()
-
-if(CMAKE_CXX_COMPILER STREQUAL clang++)
-
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
```
Comment (Twon) [inline linear_algebra/code/CMakeLists.txt]: No need to set "-Wno-unused-function" manually, as setting "-Wall" sets these flags automatically: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

File: linear_algebra/code/CMakeLists.txt [hunk 3]
```diff
@@ -48,30 +48,15 @@ include_directories(include test .)
...
-
-    set(CMAKE_C_FLAGS_RELEASE   "-O3 -march=westmere -Wall -pedantic -Wextra")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=westmere -std=c++17 -stdlib=libc++ -Wall -pedantic -Wextra -Wno-unused-parameter -Wno-unused-function")
```
Comment (Twon) [inline linear_algebra/code/CMakeLists.txt]: No need to set "-Wno-unused-parameter" manually, as setting "-Wextra" sets these flags automatically: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

File: linear_algebra/code/CMakeLists.txt [hunk 4]
```diff
@@ -48,30 +48,15 @@ include_directories(include test .)
...
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic")
-    set(CMAKE_CXX_FLAGS_DEBUG "-g -std=c++17 -Wall -pedantic -Wextra  -Wno-unused-function -Wno-unused-variable -W -Wno-unused-but-set-variable -fmax-errors=10")
+set_target_properties(la_test PROPERTIES
+    CXX_STANDARD 17
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
 
-    set(CMAKE_C_FLAGS_RELEASE   "-O3 -march=westmere -Wall -pedantic")
-    set(CMAKE_CXX_FLAGS_RELEASE "-O3 -march=westmere -std=c++17 -Wall -pedantic -Wextra -Wno-unused-function -Wno-unused-variable -W -Wno-unused-but-set-variable -fmax-errors=10")
```
Comment (Twon) [inline linear_algebra/code/CMakeLists.txt]: No need to set "-Wno-unused-but-set-variable" manually, as setting "-Wextra" sets these flags automatically: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

File: README.md [hunk 5]
```diff
@@ -1 +1,4 @@
 # wg21
+
+ [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/twon/wg21?svg=true&branch=dev_twon_appveyor)](https://ci.appveyor.com/project/twon/wg21)
```
Comment (Twon) [inline README.md]: This currently points at my Appveyor (pointing at my Github instance of the wg21 repo).  However, before this is checked in then this should be move over to point at Bob's Appveyor instance: https://ci.appveyor.com/api/projects/status/github/bobsteagall/wg21

File: linear_algebra/code/CMakeLists.txt (hunk 1 above)
Comment (BobSteagall) [reply linear_algebra/code/CMakeLists.txt]: One possible problem.  As I understand it, setting the CXX_STANDARD as 17 enables -std=g++17 which is slightly different than -std=c++17.

Comment (Twon) [reply linear_algebra/code/CMakeLists.txt]: "std=g++17" is I believe G++ with compiler-specific extensions enabled?  However, I have disabled compiler extensions explicitly with the flag:
```
    CXX_STANDARD_REQUIRED YES
    CXX_EXTENSIONS NO
```
This article describes this technique in details: https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/.  Hopefully, this should address your concerns.

File: linear_algebra/code/CMakeLists.txt [hunk 6]
```diff
@@ -48,30 +48,17 @@ include_directories(include test .)
...
-    set(CMAKE_C_COMPILER   clang)
-    set(CMAKE_CXX_COMPILER clang++)
-else()
-    set(CMAKE_C_COMPILER   gcc)
-    set(CMAKE_CXX_COMPILER g++)
-endif()
-
-if(CMAKE_CXX_COMPILER STREQUAL clang++)
-
-    set(CMAKE_C_FLAGS_DEBUG   "-g -Wall -pedantic -Wextra")
```
Comment (Twon) [inline linear_algebra/code/CMakeLists.txt]: Just a note to say explicitly setting "-g" or "-03" for debug or release is not necessary as this is the default for these compiler modes. Please see build types here: http://www.brianlheim.com/2018/04/09/cmake-cheat-sheet.html

File: linear_algebra/code/CMakeLists.txt (hunk 3 above)
Comment (Twon) [reply linear_algebra/code/CMakeLists.txt]: As discussed, can ignore comment

File: linear_algebra/code/CMakeLists.txt (hunk 4 above)
Comment (Twon) [reply linear_algebra/code/CMakeLists.txt]: As discussed, can ignore comment

File: README.md (hunk 5 above)
Comment (Twon) [reply README.md]: I've updated this now to point at Bob's instance of Appveyor.

File: linear_algebra/code/CMakeLists.txt (hunk 2 above)
Comment (Twon) [reply linear_algebra/code/CMakeLists.txt]: As discussed, my mistake, this was the "no" variant of the flag and is now added back.  Ignore comment

**Pull request merged**